salt/modules/vagrant.py

```diff
@@ -0,0 +1,466 @@
+    try:
+        vm_ = vm_cache.fetch('vagrant', name)
+    except SaltCacheError:
+        vm_ = {}
+        log.error('Trouble reading Salt cache for vagrant[%s]', name)
+    try:
+        _ = vm_['machine']
+    except KeyError:
+        raise SaltInvocationError('No Vagrant machine defined for Salt-id {}'.format(name))
+    return vm_
+
+
+def _erase_cache(name):
+    vm_cache = salt.cache.Cache(__opts__)
+    try:
+        vm_cache.flush('vagrant', name)
+    except SaltCacheError:
+        pass
+    try:
+        vm_cache.flush('vagrant_opts', name)
+    except SaltCacheError:
+        pass
+
+
+def _vagrant_ssh_config(vm_):
+    '''
+    get the information for ssh communication from the new VM
+    :param vm_: the VM's info as we have it now
+    :return: dictionary of ssh stuff
+    '''
+    machine = vm_['machine']
+    log.info('requesting vagrant ssh-config for VM %s', machine or '(default)')
+    cmd = 'vagrant ssh-config {}'.format(machine)
+    reply = __salt__['cmd.shell'](cmd,
+                                  runas=vm_.get('runas'),
+                                  cwd=vm_.get('cwd'),
+                                  ignore_retcode=True)
+    ssh_config = {}
+    for line in reply.split('\n'):  # build a dictionary of the text reply
+        tokens = line.strip().split()
+        if len(tokens) == 2:  # each two-token line becomes a key:value pair
+            ssh_config[tokens[0]] = tokens[1]
+    log.debug('ssh_config=%s', repr(ssh_config))
+    return ssh_config
+
+
+def version():
+    '''
+    Return the version of Vagrant on the minion
+
+    CLI Example:
+
+    .. code-block:: bash
+
+        salt '*' vagrant.version
+    '''
+    cmd = 'vagrant -v'
+    return __salt__['cmd.shell'](cmd)
+
+
+def list_domains():
+    '''
+    Return a cached list of all available Vagrant VMs on host.
+
+    CLI Example:
+
+    .. code-block:: bash
+
+        salt '*' vagrant.list_domains
+
+    The above shows information about all known Vagrant environments
+    on this machine. This data is cached and may not be completely
+    up-to-date.
+    '''
+    vms = []
+    cmd = 'vagrant global-status {}'
+    reply = __salt__['cmd.shell'](cmd)
+    for line in reply.split('\n'):  # build a list of the text reply
+        print(line)
+        tokens = line.strip().split()
+        try:
+            _ = int(tokens[0], 16)  # valid id numbers are hexadecimal
+            vms.append(' '.join(tokens))
+        except (ValueError, IndexError):
+            pass  # skip other lines
+    return vms
+
+
+def list_active_vms(cwd=None):
+    '''
+    Return a list of names for active virtual machine on the minion
+
+    CLI Example:
+
+    .. code-block:: bash
+
+        salt '*' vagrant.list_active_vms  cwd=/projects/project_1
+    '''
+    vms = []
+    cmd = 'vagrant status'
+    reply = __salt__['cmd.shell'](cmd, cwd=cwd)
+    for line in reply.split('\n'):  # build a list of the text reply
+        print(line)
```

Comment:
We should not ever call `print` directly from execution modules. Was this just for debugging/development?